runtime/lua/nvim/spellfile.lua

```diff
@@ -0,0 +1,278 @@
+local M = {}
+
+--- @class SpellfileConfig
+--- @field url string
+--- @field timeout_ms integer
+
+---@class SpellInfo
+---@field files string[]
+---@field key string
+---@field lang string
+---@field encoding string
+---@field dir string
+
+---@type SpellfileConfig
+M.config = {
+  url = 'https://ftp.nluug.nl/pub/vim/runtime/spell',
+  timeout_ms = 15000,
+}
+
+---@type table<string, boolean>
+M._done = {}
+
+---@return string[]
+local function rtp_list()
+  return vim.opt.rtp:get()
+end
+
+function M.isDone(key)
+  return M._done[key]
+end
+
+function M.setup(opts)
+  M._done = {}
+  M.config = vim.tbl_extend('force', M.config, opts or {})
+end
+
+local function notify(msg, level)
+  vim.notify(msg, level or vim.log.levels.INFO)
+end
+
+---@param lang string
+---@return string
+local function normalize_lang(lang)
+  local l = (lang or ''):lower():gsub('-', '_')
+  return (l:match('^[^,%s]+') or l)
+end
+
+local function writable_spell_dirs_from_rtp()
+  local dirs = {}
+  for _, dir in ipairs(rtp_list()) do
+    local spell = vim.fs.joinpath(vim.fn.fnamemodify(dir, ':p'), 'spell')
+    if vim.fn.isdirectory(spell) == 1 and vim.uv.fs_access(spell, 'W') then
+      table.insert(dirs, spell)
+    end
+  end
+  return dirs
+end
+
+local function default_spell_dir()
+  return vim.fs.joinpath(vim.fn.stdpath('data'), 'site', 'spell')
+end
+
+local function ensure_target_dir()
+  local dirs = writable_spell_dirs_from_rtp()
+  if #dirs > 0 then
+    return dirs[1]
+  end
+  local target = default_spell_dir()
```

Comment:
this should prefer a `stdpath()` dir. looking for "writable dirs in runtimepath" should be a last resort.

I assume this was matching the old logic, but makes more sense to prefer stdpath()